### PR TITLE
fix(2228): added transaction free deleteAndRelease() Method

### PR DESCRIPTION
**Pull request by sodoardi**
fixes: https://github.com/jenssegers/laravel-mongodb/issues/2228

### CODE REVIEW

**Comment by divine** [comment]
Duplicate https://github.com/jenssegers/laravel-mongodb/pull/2180

**Comment by sodoardi** [comment]
Wow.. didn't see that, looked everywhere to find a solution 🤦‍♂️
How can we get one of the fixes published? Looks like the other fix is stuck since 3 months now

**Review by alliuqemal — APPROVED** [review]

**Comment by sodoardi** [comment]
@divine any updates on how best to proceed? Close this PR and wait for the other one to be released or how do we plan on getting this bug fixed?

**Comment by divine** [comment]
> @divine any updates on how best to proceed? Close this PR and wait for the other one to be released or how do we plan on getting this bug fixed?

Hello,

I will talk with @Smolevich today and let you know what we've decided.

Thanks!

**Review by Smolevich — APPROVED** [review]

File: tests/QueueTest.php [hunk 1]
```diff
@@ -103,4 +103,20 @@ public function testIncrementAttempts(): void
...
+
+        $job = Queue::pop('test');
+        $job->release();
+
+        $jobs = Queue::getDatabase()
+            ->table(Config::get('queue.connections.database.table'))
+            ->get();
+
+        $this->assertCount(1, $jobs);
+        $this->assertEquals(1, $jobs[0]['attempts']);
```
Comment (Smolevich) [inline tests/QueueTest.php:120]: Add additional assert like [here](https://github.com/jenssegers/laravel-mongodb/pull/2180/files), please

**Review by Smolevich — CHANGES_REQUESTED** [review]

**Comment by sodoardi** [comment]
I'll take a look this weekend

**Comment by divine** [comment]
> I'll take a look this weekend

Any update?

Thanks!

**Comment by sodoardi** [comment]
> > I'll take a look this weekend
> 
> Any update?
> 
> Thanks!

@divine at least give me the whole weekend 😁
@Smolevich I added a test calling directly to the Queue::deleteAndRelease() method, I hope thats how you meant it!

**Comment by divine** [comment]
> @divine at least grant me the whole weekend 😁

Sorry but this is an open-source, contributors vanishing sometimes without any reply... and it was urgent for you too so no waiting whole weekend 🤣 

> @Smolevich I added a test calling directly to the Queue::deleteAndRelease() method, I hope thats how you meant it!

I think he meant this part:

```php
$this->assertEquals(0, $result['reserved']);
$this->assertNull($result['reserved_at']);
$this->assertEquals(
        Carbon::now()->addRealSeconds($delay)->getTimestamp(),
        $result['available_at']
);
$this->assertEquals(Carbon::now()->getTimestamp(), $result['created_at']);
$this->assertEquals($job->getRawBody(), $result['payload']);
```

Thanks!

**Comment by sodoardi** [comment]
@divine  oh no, I hope nothing bad happens to them. Should I be worried about my safety? 🤣
I did some restructuring on the tests, hope that works out! I won't be able to check back until this evening though (european time)

**Comment by Smolevich** [comment]
Resolve merge conflicts, please

**Review by Smolevich — CHANGES_REQUESTED** [review]
Needs work

**Review by divine — APPROVED** [review]

**Comment by divine** [comment]
> @divine oh no, I hope nothing bad happens to them. Should I be worried about my safety? 🤣
> I did some restructuring on the tests, hope that works out! I won't be able to check back until this evening though (european time)

Don't worry about your safety, we're just having a fun conversation, so I ruined fun now completely 😄.

I really don't know what happens with them, probably too busy and it just works 😮.

Anyways, thank you for your contribution!

**Pull request merged**